### PR TITLE
[Fuchsia] Enable sound null safety everywhere

### PR DESCRIPTION
**Pull request by zijiehe-google-com**
This change enables null_safety on fuchsia everywhere and remove the legacy conditions.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by a-siva — APPROVED** [review]

File: shell/platform/fuchsia/flutter/tests/integration/embedder/parent-view/lib/parent_view.dart [hunk 1]
```diff
@@ -181,7 +181,7 @@ Future<int> _launchChildView() async {
   final message = Int8List.fromList([0x31]);
   final completer = new Completer<ByteData>();
```
[the remark this answers is not in the document]
Comment (zijiehe-google-com) [reply shell/platform/fuchsia/flutter/tests/integration/embedder/parent-view/lib/parent_view.dart:182]: Sorry, I have no background knowledge about this test, I think using `Completer<ByteData?>` would be safer as it matches the original implementation.

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/embedding-flutter-view/lib/embedding-flutter-view.dart [hunk 2]
```diff
@@ -219,7 +219,7 @@ Future<int> _launchChildView() async {
   final message = Int8List.fromList([0x31]);
   final completer = new Completer<ByteData>();
```
[the remark this answers is not in the document]
Comment (zijiehe-google-com) [reply shell/platform/fuchsia/flutter/tests/integration/touch-input/embedding-flutter-view/lib/embedding-flutter-view.dart:220]: Done.

File: shell/platform/fuchsia/flutter/tests/integration/mouse-input/mouse-input-view/lib/mouse-input-view.dart [hunk 3]
```diff
@@ -85,13 +85,13 @@ class MyApp {
...
 
         _reportMouseInput(
-          localX: data.physicalX,
-          localY: data.physicalY,
-          buttons: data.buttons,
-          phase: data.change.name,
-          timeReceived: nowNanos,
-          wheelXPhysicalPixel: data.scrollDeltaX,
-          wheelYPhysicalPixel: data.scrollDeltaY,
+          data.physicalX,
```
[the remark this answers is not in the document]
Comment (zijiehe-google-com) [reply shell/platform/fuchsia/flutter/tests/integration/mouse-input/mouse-input-view/lib/mouse-input-view.dart:88]: Done.

**Comment by zijiehe-google-com** [comment]
Thank you Alexander and Siva. I addressed all the comments and added autosubmit label since the tree is closing now. Let me know if you have other comments.

**Pull request merged**